xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsEventExecutor.java

```diff
@@ -108,31 +107,14 @@ public void dispose()
     {
         this.jmxRegistration.unregisterMBean(MBEAN_NAME);
         for (MentionsConsumer consumer : this.consumers) {
+            // We ask the consumers to stop directly with halt, in case their is many elements in the queue.
+            // Additionally we fill the queue with stop element in order to wake up consumers and notify them
+            // to stop if the queue is empty.
             consumer.halt();
+            this.queue.add(new MentionsData().stop());
```

Comment:
Does not make much sense to create a new instance of MentionsData to return a static.